articles/151_node_to_participant_mapping.md

```diff
@@ -0,0 +1,213 @@
+  Creating more than one `Participant` increase cpu and network.
+- Each `Participant` keeps track of the others and of each DDS entity.
+  Using more than one will duplicate that data.
+- Each `Participant` creates multiple threads for event handling, discovery, etc.
+  Creating more than one will increase cpu and memory usage.
+
+For those reasons, `Participants` are heavyweight.
+
+### Current status
+
+There is a one to one map between `Nodes` and `DDS Participants`.
+This simplified the design, as `DDS Participants` provide the same organization that a `Node` needs.
+The drawback of this approach, is that performance is deteriorated.
+Furthermore, the number of `Domain participants` is limited to a small number.
+For example, [RTI connext](https://community.rti.com/kb/what-maximum-number-participants-domain) is limited to 120 participants per domain.
+
+## Proposal
+
+The goal of this proposal is to improve overall performance by avoiding the creation of one `Domain Participant` per `Node`.
+API changes will be avoided, if possible.
+
+### What is a participant mapped to?
+
+There are two main alternatives:
+- One participant per process.
+- One participant per context.
+
+The second approach allows more flexibility.
+Considering that by default there's only one context per process, it won't lower the performance.
+Moreover, a mechanism for using the same participant in two context could be added.
+
+### What is a Node now?
+
+There's not lightweight DDS entity similar to a ROS `Node`, so `Nodes` have to be built from scratch.
+A `Node` should be able to:
+- Create other entities as `Publishers`, `Subscriptions`, `Services` and `Clients`.
+  `Nodes` should own those entities, that is to say, those entity shouldn't outlive a `Node`.
+- List all its entities.
+
+For all the entities, it should be possible to get the `Node` what created them.
+Each `Participant` should store all the information needed about its nodes, and communicate it to the others.
+
+### How `Node` information is communicated?
+
+#### Using a topic
+
+The name of all the available `Nodes`, and its `Publishers`, `Subscriptions`, `Services`, `Clients` should be available for every `Participant`.
+This information can be communicated using a `topic`.
+That topic will be an implementation detail and hidden to the user.
+
+A message could be send for:
+- Each `Node`
+- Each `Participant`
+
+The second option reduces the amount of messages and it does lifetime management easier, as the message will not be more available after the `Participant` is deleted (using the correct QoS policies).
+In the following, the second option will be considered.
+
+##### State Message
+
+Each `Participant` will send a message representing their state.
+A keyed topic could be used for communicating it.
+It's reasonable to organize the data by `Participant`, so the `Participant` guid can be used as the key.
+The rest of the message will be a vector of with the information of each node.
```

Comment:
@ivanpauno I'd say `array` or `sequence` instead of `vector`, so that it matches IDL terminology.